File: lib/statuses/device_binding_statuses.dart
/// {@template device_binding_statuses}
/// This file contains the device binding statuses.
/// {@endtemplate}
enum DeviceBindingStatus {
  /// Unknown device status.
  unknown('UNKNOWN'),

  /// Failed to check the device.
  checkFail('CHECK_FAIL'),

  /// Successfully checked the device.
  checkSuccess('CHECK_SUCCESS'),

  /// Failed to bind the device.
  settingFail('SETTING_FAIL'),

  /// Successfully bound the device.
  settingSuccess('SETTING_SUCCESS'),

  /// Failed to read the device.
  readFail('READ_FAIL'),

  /// Successfully read the device.
  readSuccess('READ_SUCCESS'),

  /// Successfully check time.
  checkAndTimeSuccess('CHECK_AND_TIME_SUCCESS'),
  ;

  /// {@macro device_binding_statuses}
  const DeviceBindingStatus(this.statusValue);

  /// The value of the status.
  final String statusValue;

  /// Converts a string to a [DeviceBindingStatus].
  static DeviceBindingStatus fromString(String status) {
    return DeviceBindingStatus.values
        .firstWhere((e) => e.statusValue == status);
  }

  @override
  toString() {
    return statusValue;
  }
}

File: android/src/main/kotlin/site/shasmatic/flutter_veepoo_sdk/statuses/DeviceBindingStatuses.kt
package site.shasmatic.flutter_veepoo_sdk.statuses

/**
 * Enum class representing the status of a device binding operation.
 *
 * @constructor Creates a new [DeviceBindingStatuses] instance with the specified status.
 */
enum class DeviceBindingStatuses {

    /**
     * Unknown status.
     */
    UNKNOWN,

    /**
     * Failed to check the device.
     */
    CHECK_FAIL,

    /**
     * Successfully checked the device.
     */
    CHECK_SUCCESS,

    /**
     * Failed to bind the device.
     */
    SETTING_FAIL,

    /**
     * Successfully bound the device.
     */
    SETTING_SUCCESS,

    /**
     * Failed to read the device.
     */
    READ_FAIL,

    /**
     * Successfully read the device.
     */
    READ_SUCCESS,

    /**
     * Successfully checked and timed the device.
     */
    CHECK_AND_TIME_SUCCESS
}